frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -1,21 +1,31 @@
-import { type PropsWithChildren, useEffect, useState } from "react";
+import {
+  type PropsWithChildren,
+  createContext,
+  useEffect,
+  useState,
+} from "react";
 
 import { useSetting } from "metabase/common/hooks";
 import { setLocaleHeader } from "metabase/lib/api";
 import { loadLocalization, setUserLocale } from "metabase/lib/i18n";
+import { DatesProvider } from "metabase/ui/components/theme/DatesProvider/DatesProvider";
 
 interface LocaleProviderProps {
   locale?: string | null;
   shouldWaitForLocale?: boolean;
 }
 
+/** context for the locale used in the sdk and in public/static from the #locale parameter  */
```

Comment:
For someone like me a more detailed comment would be helpful, maybe this:

```suggestion
/** React context that retrieves the locale from the #locale parameter, and makes this locale available to the SDK, public embedding, and static embedding */
```

Reply:
Since the distinctive feature of this React Context is that it retrieves the locale from the URL rather than settings, maybe one of these names would work:

* `HashParameterLocaleContext`
* `HashLocaleContext`
* `ContextThatGetsLocaleFromURL`

To my mind, the name `FrontendLocaleContext` doesn't distinguish this component from our other locale-handling logic.

Reply:
Maybe just something like `SupplementalLocaleProvider` with a detailed explanation in jsdoc?

Reply:
That sounds good to me